crates/bevy_reflect/src/type_registry.rs

```diff
@@ -100,6 +100,23 @@ impl TypeRegistry {
             .insert(registration.type_id(), registration);
     }
 
+    /// Registers the type data `D` for type `T`.
+    ///
+    /// # Example
+    /// ```rust
+    /// use bevy_reflect::{TypeRegistry, ReflectSerialize, ReflectDeserialize};
+    ///
+    /// let mut type_registry = TypeRegistry::default();
+    /// type_registry.register::<Option<String>>();
+    /// type_registry.register_type_data::<Option<String>, ReflectSerialize>();
+    /// type_registry.register_type_data::<Option<String>, ReflectDeserialize>();
+    /// ```
+    pub fn register_type_data<T: 'static, D: TypeData + FromType<T>>(&mut self) -> Option<()> {
```

Comment:
Should we instead do this to ensure the given type can even be registered?
```suggestion
    pub fn register_type_data<T: GetTypeRegistration + 'static, D: TypeData + FromType<T>>(&mut self) -> Option<()> {
```

Reply:
> I don't think we need to even require Reflect for having something in the TypeRegistry

Correct me if I am wrong but you can only create a `TypeRegistration` via `TypeRegistration::of` which forces a `Reflect` bound on `T`

And as for registering something you can only do it via `TypeRegistry::register` which enforces a `GetTypeRegistration` bound or via `TypeRegistry::add_registration` for which you need a Reflect bound to create the registration.

Are there any plans to allow non reflect types in the registry in the future? If not I would say that requiring at least `Reflect` on this method would provide better guidance for users, at least with the current state of the public API .